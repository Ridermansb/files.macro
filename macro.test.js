const path = require("path");
const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: {
    "no usage": `import files from './macro'`,
    "correct usage": `
      import files from './macro';
      const allFiles = files('./fixtures/images');
    `,
    "no folder found": `
      import files from './macro';
      const allFiles = files('./fixtures/noFolder');
    `
  },
});