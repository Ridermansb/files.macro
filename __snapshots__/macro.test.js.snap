// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros correct usage: correct usage 1`] = `
"
import files from './macro';
const allFiles = files('./fixtures/images');

      ↓ ↓ ↓ ↓ ↓ ↓

const allFiles = \\"[ \\\\\\"calendar.jpg\\\\\\",\\\\\\"coins.jpg\\\\\\",\\\\\\"login.jpg\\\\\\",\\\\\\"menu.jpg\\\\\\",\\\\\\"passengers.jpg\\\\\\",\\\\\\"pin.jpg\\\\\\",\\\\\\"questions.jpg\\\\\\",\\\\\\"search.png\\\\\\",\\\\\\"ticket.jpg\\\\\\" ]\\";
"
`;

exports[`macros no files: no files 1`] = `
"
import files from './macro';
const allFiles = files('./fixtures/noFiles');

      ↓ ↓ ↓ ↓ ↓ ↓

const allFiles = \\"[  ]\\";
"
`;

exports[`macros no folder found: no folder found 1`] = `
"
import files from './macro';
const allFiles = files('./fixtures/noFolder');

      ↓ ↓ ↓ ↓ ↓ ↓

const allFiles = \\"[]\\";
"
`;

exports[`macros no usage: no usage 1`] = `
"
import files from './macro'

      ↓ ↓ ↓ ↓ ↓ ↓


"
`;
